File: Backend-sistem-SP/src/main/java/backend/security_alert/dto/zona/ZonaResponse.java
package backend.security_alert.dto.zona;

import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ZonaResponse {

    private Long id;
    private String nombre;
    private String descripcion;
    private String ubicacion;
    private String estadoActual;
    private String modoControl;
    private Boolean activa;
    private List<SensorResponse> sensores;
}

File: Frontend-sistema-SP/src/types/requestType/zona/ZonaResponse.ts
import type { SensorResponse } from "./SensorResponse";
import type { ProgramacionHoraria } from "./ProgramacionHoraria";

export type ZonaResponse = {
    id: number;
    dispositivoId?: number;
    nombre: string;
    descripcion?: string;
    ubicacion?: string;
    estadoActual?: string;
    modoControl?: string;
    activa: boolean;
    sensores?: SensorResponse[];
    programaciones?: ProgramacionHoraria[];
}
